doc/src/pair_lj_sphere.rst

```diff
@@ -0,0 +1,130 @@
+.. index:: pair_style lj/sphere
+.. index:: pair_style lj/sphere/omp
+
+pair_style lj/sphere command
+============================
+
+Accelerator Variant: *lj/sphere/omp*
+
+Syntax
+""""""
+
+.. code-block:: LAMMPS
+
+   pair_style style args
+
+* style = *lj/sphere*
+* args = list of arguments for a particular style
+
+.. parsed-literal::
+
+     *lj/sphere* args = cutoff ratio
+       cutoff = global cutoff ratio for Lennard Jones interactions (unitless)
+
+Examples
+""""""""
+
+.. code-block:: LAMMPS
+
+   pair_style lj/sphere 2.5
+   pair_coeff * * 1.0
+   pair_coeff 1 1 1.1 2.8
+
+Description
```

Comment:
This section needs some context like we disccused.  About using it for mildly polydisperse systems for efficiency reasons. Also advice to bin particle sizes by type, if want to use it with neigh multi. 
Also, the dangers of using a cutoff scale factor less than 2*(1/6) which leads to discontinuous forces. Possibly that should be a warning or error in the code.

Reply:
For atom sphere we added the `neigh_modify collection/interval` command to simplify the multi style. This option requires a pair style have the `atom2cut()` and `radii2cut()` so the neighbor class can sort atoms into collections and determine the stencil size for each collection bin. The flag `force->pair->finitecutflag` is also required.

In the granular package, these are defined here:
https://github.com/lammps/lammps/blob/cbf7f66a6f48080fe01ac2b919ed22474a2bdb08/src/GRANULAR/pair_gran_hooke_history.cpp#L815
https://github.com/lammps/lammps/blob/cbf7f66a6f48080fe01ac2b919ed22474a2bdb08/src/GRANULAR/pair_granular.cpp#L848

That being said, I think the regular per-type multi should still work without these methods. So it's odd a significant speedup wasn't attained...

Reply:
> That being said, I think the regular per-type multi should still work without these methods. So it's odd a significant speedup wasn't attained...

Feel free to experiment. I am attaching my test example.
[in.melt-sphere-random.txt](https://github.com/lammps/lammps/files/11145084/in.melt-sphere-random.txt)


> For atom sphere we added the `neigh_modify collection/interval` command to simplify the multi style.

Before I can implement those correctly, I have to spend more time reading the pair styles' that use those.

Reply:
I would propose to try and finish this PR "as is" and then try improvements together with the version for granular. For the granular style I can even see two options: one like lj/cut where sigma is taken from the atom diameter and a second like lj/expand, where sigma is specified (and the same for the same atom type) and instead the "delta" parameter is taken from the atom diameter. In the context of DEM models, the latter makes more sense to me.
This pair style is more like a shortcut to set up lj/cut style simulations with 100s or 1000s of atom types (as in the one example I provided on MatSci.org).

Reply:
> I don't think multi will perform well for this example since radii are uniformly distributed.

I've modified it and the corresponding example in the docs to create a bimodal distribution. Now "multi" is about 30% faster than "bin".
![snap](https://user-images.githubusercontent.com/289251/229950967-50a22c65-070f-48c8-9ecb-be24e5ddd236.jpg)